src/main/java/com/everis/control/PersistenceService.java

```diff
@@ -1,45 +1,72 @@
 package com.everis.control;
 
+import java.util.ArrayList;
 import java.util.List;
-
-
+import java.util.Map;
 
 import javax.ejb.Stateless;
 import javax.persistence.EntityManager;
 import javax.persistence.PersistenceContext;
 import javax.persistence.TypedQuery;
-
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Expression;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
 
 @Stateless
-public class PersistenceService<T,L> {
+public class PersistenceService<T, L> {
 
 	@PersistenceContext(unitName = "car-unit")
 	private EntityManager em;
 
-	
-	public List<T> getEntitiesWithNamedQuery(String namedQuery, Class<T> c) {
-		TypedQuery<T> query = em.createNamedQuery(namedQuery, c);
-		return query.getResultList();
-	}
+	public TypedQuery<T> getEntitiesQuery(Class<T> c, Map<String, String> filterMap, String orderBy) {
+
+		CriteriaBuilder criteriaBuilder = em.getCriteriaBuilder();
+		CriteriaQuery<T> criteriaQuery = criteriaBuilder.createQuery(c);
+		Root<T> root = criteriaQuery.from(c);
+		criteriaQuery.select(root);
+
+		List<Predicate> predicates = new ArrayList<Predicate>();
+
+		for (Map.Entry<String, String> entry : filterMap.entrySet()) {
+			Expression<String> expression = criteriaBuilder.lower(root.get(entry.getKey()).as(String.class));
+			String value = String.format("%%%s%%", entry.getValue().toLowerCase());
+			predicates.add(criteriaBuilder.like(expression, value));
+		}
 
+		Predicate predicate = criteriaBuilder.or(predicates.toArray(new Predicate[0]));
+		criteriaQuery.where(predicate);
+
+		if (orderBy != null && !orderBy.isEmpty()) {
+			if (orderBy.charAt(0) == '-') {
```

Comment:
Despite of the emptiness of orderBy, I do not see any checking for its content, following the logic looks like the value can be anything or anything starting with - . Could you please check it ?